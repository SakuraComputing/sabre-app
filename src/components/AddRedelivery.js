import React from 'react';
import { connect } from 'react-redux';
import RedeliveryForm from './RedeliveryForm';
import { createRedelivery } from "../actions/redelivery";
import UrlConstants from '../../config/url-constants';

export class AddRedelivery extends React.Component {
    onSubmit = (redelivery) => {
      this.props.createRedelivery(redelivery);
      this.props.history.push(UrlConstants.REDELIVERY_HOME);
    };
    render() {
        return (
            <div>
                <h2>Add Redelivery</h2>
                <RedeliveryForm
                    onSubmit = {this.onSubmit}
                />
            </div>
        );
    }
}

export const mapDispatchToProps = (dispatch) => ({
    createRedelivery: (redelivery) => dispatch(createRedelivery(redelivery))
});
export default connect(undefined, mapDispatchToProps)(AddRedelivery);