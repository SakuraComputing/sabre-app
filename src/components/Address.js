import React from 'react';

const Address = ( { address } ) => (
    <table className='address-panel'>
        <tbody>
            <tr>{address.addressName}</tr>
            <tr>{address.addressLine1}</tr>
            <tr>{address.addressLine2}</tr>
            <tr>{address.town}</tr>
            <tr>{address.county}</tr>
            <tr>{address.telephone}</tr>
        </tbody>
    </table>
);
export default Address;
