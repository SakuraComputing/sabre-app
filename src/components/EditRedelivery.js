import React from 'react';
import { connect } from 'react-redux';
import { updateRedelivery, deleteRedelivery} from '../actions/redelivery';
import RedeliveryForm from './RedeliveryForm';
import UrlConstants from "../../config/url-constants";

const REDELIVERY_URL = UrlConstants.REDELIVERY_ID;

export class EditRedelivery extends React.Component {

    onSubmit = (redelivery) => {
        this.props.updateRedelivery(this.props.redelivery.id, redelivery);
        this.props.history.push('/redelivery');
    };

    onDelete = (id) => {
        this.props.deleteRedelivery( this.props.cardId);
        this.props.history.push('/redelivery');
    };

    render() {
        return (
            <div>
                <h2>Edit Redelivery</h2>
                <RedeliveryForm
                    redelivery={this.props.redelivery}
                    onSubmit={this.onSubmit}
                />
                <button
                    onClick={this.onDelete}>
                    Delete Redelivery
                </button>
            </div>
        );
    }
}

export const mapStateToProps = ( state, props ) => ({
   redelivery: state.redeliveries.find((redelivery) => {
       return redelivery.cardId === props.match.params.id;
   })
});

export const mapDispatchToProps = (dispatch) => ({
    updateRedelivery: (id, redelivery) => dispatch(updateRedelivery(id, redelivery)),
    deleteRedelivery: (data) => dispatch(deleteRedelivery(data))
});
export default connect(mapStateToProps, mapDispatchToProps)(EditRedelivery);
