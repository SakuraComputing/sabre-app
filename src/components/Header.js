import React from 'react';
import { NavLink } from 'react-router-dom';
import { Navbar } from 'react-bootstrap';

const Header = () => (
    <div className='menu-container'>
        <Navbar className="navbar navbar-default navbar-static-top" >
            <ul>

                <li>
                    <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
                </li>
                <li>
                    <NavLink to="/tracking" activeClassName="is-active">Parcel Tracking</NavLink>
                </li>
                <li>
                    <NavLink to="/redelivery" activeClassName="is-active">Redelivery</NavLink>
                </li>
            </ul>
        </Navbar>
        <h1>Yodel Web Portal</h1>
    </div>
);
export default Header;
