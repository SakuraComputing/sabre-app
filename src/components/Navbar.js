import React from 'react';

export default class Navbar extends React.Component {

    constructor(props) {
        super(props);
    }

    onHamburgerClick = () =>  {
      this.props.onHamburgerClick && this.props.onHamburgerClick();
    };

    render(){
        return (
            <nav className="navbar navbar-default navbar-fixed-top">
                <div className="container-fluid">
                    <div className="navbar-header">
                        <button type="button" className="navbar-btn navbar-toggle collapsed" onClick={this.onHamburgerClick}>
                            <span className="sr-only">Toggle Navigation</span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                        </button>
                    </div>
                    {this.props.children}
                </div>
            </nav>
        );
    }
}