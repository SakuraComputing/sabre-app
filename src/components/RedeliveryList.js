import React from 'react';
import { connect } from 'react-redux';
import RedeliveryItem from '../components/RedeliveryItem';
import selectRedeliveries from './../selectors/Redeliveries';

export const RedeliveryList = (props) => (
    <table>
        <tbody>
            <tr>
                <th>Card Id</th>
                <th>Parcel Id</th>
                <th>Address</th>
                <th>Latest Scan</th>
                <th>Request Date</th>
            </tr>
            {
                props.redeliveries.map((redelivery) => {
                    return <RedeliveryItem key={redelivery.id} {...redelivery}/>
                })
            }
        </tbody>
    </table>
);

export const mapStateToProps = (state) => {
    return {
        redeliveries: selectRedeliveries(state.redeliveries, state.filters)
    };
};

export default connect(mapStateToProps)(RedeliveryList);


