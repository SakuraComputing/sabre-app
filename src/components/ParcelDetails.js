import React from 'react';
import Address from "./Address";

const ParcelDetails = ({ parcel, address }) => (
    <table className='parcel-details' frameBorder="1">
        <tbody>
            <tr>Address:</tr>
            <tr className="address-details" >{address ? <Address address={address}/> : 'Address not found'}</tr>
            <tr>Service Code: {parcel.serviceCode}</tr>
        </tbody>
    </table>
);
export default ParcelDetails;
