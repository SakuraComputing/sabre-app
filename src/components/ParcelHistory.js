import React from 'react';
import parcelData from '../../data/parcel.json';
import trackData from '../../data/tracks.json';
import addressData from '../../data/address.json';
import ParcelDetails from "./ParcelDetails";
import TrackingHistory from "./TrackingHistory";

class ParcelHistory extends React.Component {

    constructor(props) {
        super(props);
        this.state = {parcelId: '', error: '', selectedParcel: '', selectedTracks: '', selectedAddress: ''};
    }

    onSubmit = (e) => {
        e.preventDefault();

        let parcelId = this.state.parcelId;
        if (parcelId) {
            if(this.onValidate()) {
                const selectedParcel = parcelData["parcels"].find(parcel => parcel.parcelId === parcelId);
                if (selectedParcel) {
                    let addressNo = selectedParcel.addressNo;
                    var selectedAddress = addressData["addresses"].find(address => address.addressNo === addressNo);
                    var selectedTracks = trackData["tracks"].filter(track => track.parcelId === parcelId);
                }
                let error = selectedParcel ? null : 'parcel not found';

                this.setState(() => ({
                    selectedParcel: selectedParcel,
                    error: error,
                    selectedTracks: selectedTracks,
                    selectedAddress: selectedAddress
                }));
            }
        } else {
            this.setState(() => ({ error: 'Please provide a parcel Id' }));
        }
    };

    onParcelChange = (e) => {
        const parcelId = e.target.value;
        this.setState(() => ({ parcelId }));
    };

    onClear = () => {
        this.setState(()=> ({ parcelId: '', error: null, selectedParcel: '', selectedTracks: '', selectedAddress: '' }))
    };

    onValidate = (e) => {
        let parcelId = this.state.parcelId;
        let formIsValid = true;

        if(parcelId.length < 12 || parcelId.length > 35) {
            this.setState(() => ({ error: 'Parcel must be between 12 and 35 chars' }));
            formIsValid = false
        }
        return formIsValid;
    }
    ;

    render() {
        let id = 'parcelId';
        return (
            <div>
                <form className="quick-form" onSubmit={this.onSubmit} >
                    <h2>Parcel History</h2>
                    <div className="search-bar">
                        Parcel Id:
                        <input
                            onChange={this.onParcelChange}
                            name={id}
                            id={id}
                            value={this.state.parcelId}
                            placeholder="Parcel Id"
                        />
                        <button className="submit-button">
                            Search
                        </button>
                        <button type="reset" className="clear-button" onClick={this.onClear}>
                            Clear Search
                        </button>
                    </div>
                    {this.state.selectedParcel ? <ParcelDetails parcel={this.state.selectedParcel} address={this.state.selectedAddress}/> : ''}
                    {this.state.selectedTracks ? <TrackingHistory tracks={this.state.selectedTracks} /> : ''}
                </form>
                {this.state.error && <div className="error-message">{this.state.error}</div>}
            </div>
        )
    };
};
export default ParcelHistory;

