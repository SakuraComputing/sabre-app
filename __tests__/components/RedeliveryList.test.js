import React from 'react';
import {mapStateToProps, RedeliveryList} from '../../src/components/RedeliveryList';
import { shallow, mount } from 'enzyme';
import redeliveries from "../fixtures/redeliveries";

describe('Redelivery List Component',() => {
   it('should render the form correctly',() => {
        const testComponent = shallow(<RedeliveryList redeliveries={redeliveries}/>);
        expect(testComponent).toMatchSnapshot();
   });
   it('should display no redeliveries when nothing is passed in',() => {
       const testComponent = shallow(<RedeliveryList redeliveries={[]}/>);
       expect(testComponent).toMatchSnapshot();

   });
    // it('maps state to props',() => {
    //     const currentState = {
    //         cardId: '',
    //         parcelId: '',
    //         latestScan: 0,
    //         requestDate: 0
    //     };
    //     const redeliveriesTest = {
    //         id: 1,
    //         cardId: '74747',
    //         parcelId: 'JD000',
    //         latestScan: 0,
    //         requestDate: 0
    //     };
    //     let state = {
    //         redeliveries: { redeliveriesTest },
    //         filters: currentState
    //     };
    //     expect(mapStateToProps(state).redeliveries).toEqual(state);
    // });
});


