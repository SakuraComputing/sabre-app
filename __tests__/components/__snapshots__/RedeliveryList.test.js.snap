// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redelivery List Component should display no redeliveries when nothing is passed in 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Card Id
      </th>
      <th>
        Parcel Id
      </th>
      <th>
        Address
      </th>
      <th>
        Latest Scan
      </th>
      <th>
        Request Date
      </th>
    </tr>
  </tbody>
</table>
`;

exports[`Redelivery List Component should render the form correctly 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Card Id
      </th>
      <th>
        Parcel Id
      </th>
      <th>
        Address
      </th>
      <th>
        Latest Scan
      </th>
      <th>
        Request Date
      </th>
    </tr>
    <RedeliveryItem
      addressNo={5}
      cardId="876363"
      id="1"
      key="1"
      latestScan={1262304000000}
      parcelId="JD0002289756473452"
      requestDate={1262304000000}
    />
    <RedeliveryItem
      addressNo={6}
      cardId="876002"
      id="2"
      key="2"
      latestScan={1293840000000}
      parcelId="JD0002289756470001"
      requestDate={1293840000000}
    />
    <RedeliveryItem
      addressNo={7}
      cardId="376363"
      id="3"
      key="3"
      latestScan={1325376000000}
      parcelId="JD0002289756470005"
      requestDate={1325376000000}
    />
  </tbody>
</table>
`;
